src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3832,6 +3833,15 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
             return _syntaxFactory.Parameter(attributes, modifiers.ToList(), type, name, exclamation, def);
         }
 
+        private SyntaxToken AddLanguageVersionCheckToNullCheckedParameter(SyntaxToken exclamation)
+        {
+            if (this.Options.LanguageVersion < LanguageVersion.CSharp8 && !(exclamation is null))
```

Comment:
Please use CheckFeatureAvailability instead of checking the language version directly here, in the vein of other feature availability checks in LanguageParser.cs